### ISSUE TITLE
Duration using decimal hours produces wrong output using toFormat

### ISSUE THREAD

**Issue by fooey** [moment/luxon#1070]
**Describe the bug**

Creating a duration using a decimal hour then using `toFormat` produces an incorrect result

The inputs we've seen wrong so far are `2.4` and `5.3`

**To Reproduce**

`luxon.Duration.fromObject({hour: 2.4}).toFormat('hh:mm')` returns `"02:23"`
`luxon.Duration.fromObject({hour: 5.3}).toFormat('hh:mm')` returns `"05:17"`

**Actual vs Expected behavior**

2.4 hours is 02:24
5.3 hours is 05:18

**Desktop (please complete the following information):**

- OS: Windows11
- Browser: Firefox 93
- Luxon version: 2.0.2
- Your timezone: "America/Denver"

**Additional context**

_Converting to different units from hours produces correct results_

`luxon.Duration.fromObject({minute: 2.4 * 60}).toFormat('hh:mm')` = `"02:24"`
`luxon.Duration.fromObject({second: 2.4 * 60 * 60}).toFormat('hh:mm')` = `"02:24"`
`luxon.Duration.fromObject({hour: 2.4}).shiftTo('seconds').toFormat('hh:mm')` = `"02:24"`
`luxon.Duration.fromMillis(8640000).toFormat('hh:mm')` = `"02:24"`

_Shifting to hours also produces the wrong result_

`luxon.Duration.fromObject({minute: 2.4 * 60}).shiftTo('hour').toFormat('hh:mm')` = `"02:23"`

_The durations still produce the same valueOf_

`luxon.Duration.fromObject({hour: 2.4}).valueOf()` = `8640000`
`luxon.Duration.fromObject({minute: 2.4 * 60}).valueOf()` = `8640000`

**Comment by icambron** [comment]
This is some kind of floating point math problem:

```js
Duration.fromObject({hour: 2.4}).shiftTo("hours", "minutes").toObject() //=> { hours: 2, minutes: 23.999999999999993 }
```

Then `mm` formats 23 because it's not quite 24. I'd definitely accept a fix for this

**Comment by bluSCALE4** [comment]
Addressed by #1078

**Comment by icambron** [comment]
@bluSCALE4 Cool. I'll get this into the next version. I'll probably add a few more test cases just to make sure it doesn't break anything new

**Comment by BulatSa** [comment]
Hello, i have some issue 
`Duration.fromObject({hour: 0, minute: 10}).toFormat('HH:mm') returns "HH:10"`
or
`Duration.fromMillis(60000).toFormat('HH:mm') returns "HH:10"`

**Comment by icambron** [comment]
Fixed in 2.2.0
